source/postprocess/visualization/shear_stress_eigenvectors.cc

```diff
@@ -0,0 +1,340 @@
+
+            const double tau_magnitude = 0.5 * (eigenvector_array[0].first - eigenvector_array[2].first);
+
+
+            Point<dim> p = in.position[q];
+            double x = p[0];
+            double y = p[1];
+            double z = p[2];
+            double e = 1.1e-1;
+            Tensor<1,dim> ev0 = eigenvector_array[0].second;
+            Tensor<1,dim> ev1 = eigenvector_array[1].second;
+            Tensor<1,dim> ev2 = eigenvector_array[2].second;
+            //if(fabs(x-0.5) < e && fabs(y-0.25) < e && fabs(z-0.375) < e || fabs(x-0.625) < e && fabs(y-0.375) < e && fabs(z-0.5) < e)
+            {
+              //std::cout << x << "," << y << "," << z << ", strain rate = " << compressible_strain_rate << std::endl;
+              //std::cout << x << "," << y << "," << z << ": tau_magnitude = " << tau_magnitude <<  ", eigenvector 0 = " << eigenvector_array[0].second << ", 1 = " << eigenvector_array[1].second<< ", 2 = " << eigenvector_array[2].second << " = " << tau_direction << ", norm = " << tau_direction/tau_direction.norm() << std::endl;
+              std::cout << x << " " << y << " " << z << " "
+                        << eigenvector_array[0].first << " "<< eigenvector_array[1].first << " "<< eigenvector_array[2].first << " "
+                        << ev0 << " " << ev1 << " " << ev2
+                        << tau_direction << std::endl;
+            }
+
+            for (unsigned int i=0; i<dim; ++i)
+              {
+                // the eigenvectors are unit
+                AssertThrow(std::fabs(eigenvector_array[0].second.norm()-1) < 1e-8,ExcMessage("eigenvector array 0 norm is not one: " + std::to_string(eigenvector_array[0].second.norm())));
+                AssertThrow(std::fabs(eigenvector_array[1].second.norm()-1) < 1e-8,ExcMessage("eigenvector array 1 norm is not one: " + std::to_string(eigenvector_array[1].second.norm())));
+                AssertThrow(std::fabs(eigenvector_array[2].second.norm()-1) < 1e-8,ExcMessage("eigenvector array 2 norm is not one: " + std::to_string(eigenvector_array[2].second.norm())));
+                Tensor<1,dim> scaled_eigenvector = eigenvector_array[i].second * eigenvector_array[i].first;
+                for (unsigned int j=0; j<dim; ++j)
+                  {
+                    computed_quantities[q](i*dim+j) = scaled_eigenvector[j];
+                  }
+              }
+
+            for (unsigned int i=0; i<dim; ++i)
+              computed_quantities[q](dim*dim+i) = eigenvector_array[i].first;
+
+          }
+      }
+
+
+      template <int dim>
+      std::vector<std::string>
+      ShearStressEigenvectors<dim>::get_names () const
+      {
+        std::vector<std::string> names;
+        switch (dim)
+          {
+            case 2:
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
+              names.emplace_back("shear_stress_eigenvectors_sigma_1");
```

Comment:
is this a mistake? shouldn't these only be repeated twice instead of three times? same for the next three lines, of course